include/dlaf/matrix/copy_tile.h

```diff
@@ -134,6 +162,28 @@ void copy(TileElementSize region, TileElementIndex in_idx, const Tile<const T, D
 
 DLAF_MAKE_CALLABLE_OBJECT(copy);
 
+// copy overload taking a policy and a sender, returning a sender. This can be
+// used in task graphs.
+template <Backend B, typename Sender,
+          typename = std::enable_if_t<hpx::execution::experimental::is_sender_v<Sender>>>
+auto copy(const dlaf::internal::Policy<B> p, Sender&& s) {
+  return dlaf::internal::transform<B>(p, copy_o, std::forward<Sender>(s));
+}
+
+// copy overload taking a policy, returning a partially applied algorithm. This
+// can be used in task graphs with the | operator.
+template <Backend B>
+auto copy(const dlaf::internal::Policy<B> p) {
+  return dlaf::internal::PartialTransform{p, copy_o};
+}
+
+// copy overload taking a policy and plain arguments. This is a blocking call.
+template <Backend B, typename T1, typename T2>
+void copy(const dlaf::internal::Policy<B> p, T1&& t1, T2&& t2) {
+  hpx::execution::experimental::sync_wait(
+      copy(p, hpx::execution::experimental::just(std::forward<T1>(t1), std::forward<T2>(t2))));
+}
+
```

Comment:
Actually, yes, I think so! Good catch. I think this predates the macro. I'll see if there was another reason not to use the macro.

Reply:
One of the overloads was slightly different, but it was inconsequential as it was unused. I've updated this to use the macro.

Reply:
Nice! Just one question: what is the rationale of moving it to `internal`?

Reply:
1. It's consistent with the other tile algorithms.
2. It's the "unsafe" version of copy. For host-side copies it's all the same since the copy is a blocking call, but the device-side `internal::copy` is the version that takes a stream and does no synchronization. This should only be used in special cases (like in the implementations of the sender overloads) or if you know you're going to synchronize manually at the right point. The sender overloads expose the "safe" asynchronous version, i.e. the one that returns a sender that should be `sync_wait`ed or connected to some other algorithm.